solvers/skglm.py

```diff
@@ -28,22 +31,18 @@ def set_objective(self, X, y, lmbd):
         n_samples = self.X.shape[0]
 
         warnings.filterwarnings('ignore', category=ConvergenceWarning)
-        self.logreg = SparseLogisticRegression(
-            alpha=self.lmbd / n_samples, max_iter=1, max_epochs=50_000,
-            tol=1e-12, fit_intercept=False, warm_start=False, verbose=False)
+        self.l1_penalty = compiled_clone(L1(lmbd / n_samples))
```

Comment:
Can you use a GeneralizedLinearEstimator so that the compilation job is done by the estimator at fit time instead ? IMO it's clearer if we don't expose the whole mechanism